File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/Conditional.java
// Conditional.java

import java.io.PrintWriter;

public class Conditional extends Ternary
{
	public Conditional(Expression aOperand1, Expression aOperand2, Expression aOperand3)
	{
		super(aOperand1, aOperand2, aOperand3);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputInfix(aPrintWriter, "?", ":");
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/conditional.cpp
// conditional.cpp

#include <iostream>
#include "conditional.h"
#include "dp_exception.h"
class expression;
#include "ternary.h"

conditional::conditional(expression * a_operand1, expression * a_operand2, expression * a_operand3)
: ternary(a_operand1, a_operand2, a_operand3)
{
}

void conditional::output(std::ostream & a_ostream) const
{
	output_infix(a_ostream, "?", ":");
}
